### ISSUE TITLE
Questions about Maxnorm regularizations

### ISSUE THREAD

**Issue by chenbinghui1** [ShadeAlsha/LTR-weight-balancing#5]
In paper, it claims that Maxnorm regularization is applied per iteration, but in the code, it seems to be applied per epoch? Am i right?

**Comment by ShadeAlsha** [comment]
In the demo, MaxNorm is applied in each epoch as you said. This corresponds to lines 140-142 in trainval.py. 

https://github.com/ShadeAlsha/LTR-weight-balancing/blob/0e9494cad5b4805642f05097a336096f780300ee/utils/trainval.py#L140-L142

You can move this code up in order to apply it per-iteration.

**Comment by rahulvigneswaran** [comment]
Looks like there is a significant difference in performance between the two.

<img width="481" alt="Screenshot 2022-07-01 at 9 35 29 AM" src="https://user-images.githubusercontent.com/37276766/176821398-f2e09ef8-669a-4cb7-991a-a9c2ca57de7c.png">

@ShadeAlsha @chenbinghui1

**Comment by ShadeAlsha** [comment]
Yes, PGD is supposed to be applied in each iteration as in page 4 of the paper and as chenbinghui1 mentioned earlier :)
I haven't updated the code yet, but in the mean time your modified version should be good if you just moved PGD to the iteration loop.
Thank you.

**Comment by xiaohe6** [comment]
Hello, I would like to ask, when I used the author's source code to fine-tune, how is it that the addition and removal of maximum regularization are the same? Does it seem to not work?
